url71665d7d68_branscha_lib_scripty_tgz-pJ8-com_sdicons_scripty_ScriptyAutoReplJ8/src/com/sdicons/scripty/cmdlib/BeanLibrary.java

```diff
@@ -0,0 +1,560 @@
+                Class[] lClasses = new Class[lArgs.length];
+                for(int i = 0; i < lArgs.length; i++) lClasses[i] = (lArgs[i]==null)?null:lArgs[i].getClass();
+
+                lMeth = lTarget.getClass().getMethod((String) aArgs[2], lClasses);
+            }
+
+            if(lMeth == null)
+                throw new CommandException("Method could not be resolved.");
+
+            return lMeth.invoke(lTarget, lArgs);
+        }
+        catch (Exception e)
+        {
+            throw new CommandException("Invocation failed.\n" + e.getMessage(), e);
+        }
+    }
+
+    // Update the properties of a bean.
+    // (bean-upd BEAN prop1=val1 prop2=val2 ...)
+    //
+    @ScriptyCommand(name="bean-upd")
+    public void beanUpd(Object[] aArgs, IContext aCtx)
+    throws CommandException
+    {
+        final String lPath = guardStringOther(aArgs);
+        final Resolution lRes = resolve(lPath, aCtx);
+        if(! filter(lRes)) throw new CommandException("Cannot update filtered.");
+        for(int i = 2; i < aArgs.length; i++)
+        {
+            if(!(aArgs[i] instanceof Pair))
+                throw new CommandException("Pair expected.");
+            Pair lPair = (Pair) aArgs[i];
+            update(lRes.getVal(), lPair.getLeft(), lPair.getRight());
+        }
+    }
+
+    private String guardStringOther(Object[] aArgs)
+    throws CommandException
+    {
+        if(aArgs.length < 3)
+            throw new CommandException("ERROR - Too few args.");
+        if(!(aArgs[1] instanceof String))
+            throw new CommandException("ERROR - expected a string as first arg.");
+        return (String) aArgs[1];
+    }
+
+    public static interface IFilter
+    {
+        boolean filter(String aPath, Object aValue);
+    }
+
+    private IFilter[] filters = new IFilter[]{};
+
+    private static Pattern PAT_FIELDNAME_BIS =  Pattern.compile("^\\s*([^\\s\\[\\]]+)?\\s*(\\[\\s*([^\\s\\[\\]]+)\\s*\\])*\\s*$");
+
+    private static class Resolution
+    {
+        private Object val;
+        private String path;
+
+        public Resolution(String aPath, Object aObj)
+        {
+            path = aPath;
+            val = aObj;
+        }
+
+        public String getPath()
+        {
+            return path;
+        }
+
+        public Object getVal()
+        {
+            return val;
+        }
+    }
+
+    public Resolution resolve(String aPath, IContext aCtx)
+    throws CommandException
+    {
+        if(aPath.startsWith("/"))
+        {
+            final String lRelPath = aPath.substring(1);
+            List<String> lPieces = canonize(lRelPath);
+
+            Object lVal;
+            if("".equals(lRelPath)) lVal = aCtx;
+            else lVal = resolveCanonical(lPieces, 0, aCtx);
+
+            final StringBuilder lBuilder = new StringBuilder("/");
+            for(String lPart : lPieces)
+            {
+                lBuilder.append(lPart);
+                if(lPieces.get(lPieces.size() - 1) != lPart)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.